### PR TITLE
feat: open create item screen

### PR DESCRIPTION
**Pull request by juandgaines**
- NAvigation from agenda to agenda item.
- First appraoch viemodel agenda item
image

### CODE REVIEW

File: agenda/data/src/main/java/com/juandgaines/agenda/data/mappers/ReminderMappers.kt [hunk 1]
```diff
@@ -33,7 +33,7 @@ fun ReminderResponse.toReminder() = Reminder(
 fun ReminderEntity.toReminder() = Reminder(
     id = id,
     title = title,
-    description = description,
+    description = description?: "",
```
Comment (Kai-Cilliers) [inline agenda/data/src/main/java/com/juandgaines/agenda/data/mappers/ReminderMappers.kt:36]: You can also use this neat Kotlin extension function:

```
description = description.orEmpty()
```

File: agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/agenda_item/AgendaItemViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+
+    private val navParameters = savedStateHandle.toRoute<ScreenNav.AgendaItem>()
+
+    init {
+        viewModelScope.launch {
+            val idItem = navParameters.id
+            state = state.copy(
+                isEditing = navParameters.isEditing
+            )
```
Comment (Kai-Cilliers) [inline agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/agenda_item/AgendaItemViewModel.kt:51]: There have been some interesting discussions recently related to triggering logic required when a user lands on a particular screen, which I think you may find interesting.

Philipp has made a video covering this topic. If you prefer reading, I've also linked some great articles.

1. Phlipp's video: https://www.youtube.com/watch?v=mNKQ9dc1knI
2. https://proandroiddev.com/loading-initial-data-in-launchedeffect-vs-viewmodel-f1747c20ce62
3. https://proandroiddev.com/loading-initial-data-part-2-clear-all-your-doubts-0f621bfd06a0

File: agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/agenda_item/AgendaItemViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            SelectDateStart -> TODO()
+            SelectTimeStart -> TODO()
+            Delete -> TODO()
+            Edit -> {
+                state = state.copy(
+                    isEditing = true
+                )
+            }
+            Save -> TODO()
+            else -> Unit
```
Comment (Kai-Cilliers) [inline agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/agenda_item/AgendaItemViewModel.kt:111]: I tend to be cautious when adding an `else` in a `when` statement. When you do, the compiler won’t warn you that you haven’t implemented a specific action when adding a new one. In this specifc scenarion I like to do define each branch, something similar to this:

```
when (action) {
    Edit -> {
      state = state.copy(
        isEditing = true
      )
    }
    /* Handled at navigation level */
    is Save -> Unit
    is ExampleActionOne -> Unit
    is ExampleActionTwo -> Unit
  }
```

Again, just a preference of mine 😄

**Review by Kai-Cilliers — APPROVED** [review]

File: agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/agenda_item/AgendaItemViewModel.kt (hunk 3 above)
Comment (juandgaines) [reply agenda/presentation/src/main/java/com/juandgaines/agenda/presentation/agenda_item/AgendaItemViewModel.kt]: dione

**Pull request merged**